src/js/utils.js

```diff
@@ -482,18 +482,39 @@ export function trimValue(val) {
     return result;
 }
 
-export function calculateMidColor(c1, c2) {
-    let c = "#";
-    for (let i = 0; i < 3; i++) {
-        let sub1 = c1.substring(1 + 2 * i, 3 + 2 * i);
-        let sub2 = c2.substring(1 + 2 * i, 3 + 2 * i);
-        let v1 = parseInt(sub1, 16);
-        let v2 = parseInt(sub2, 16);
-        let v = Math.floor((v1 + v2) / 2);
-        let sub = v.toString(16).toUpperCase();
-        let padsub = ('0' + sub).slice(-2);
-        c += padsub;
+export function calculateThemeBackgroundColor(iconColor){
+    const opacity = '0.2';
+    let colorRgb = hexToRgb(iconColor);
+
+    return rgba2hex(`rgb(${colorRgb.r}, ${colorRgb.g}, ${colorRgb.b}, ${opacity})`);
+}
+
+export function hexToRgb(hex) {
+    let result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
+    return result ? {
+        r: parseInt(result[1], 16),
+        g: parseInt(result[2], 16),
+        b: parseInt(result[3], 16)
+    } : null;
+}
+
+export function rgba2hex(orig) {
+    let a,
+        rgb = orig.replace(/\s/g, '').match(/^rgba?\((\d+),(\d+),(\d+),?([^,\s)]+)?/i),
+        alpha = (rgb && rgb[4] || "").trim(),
+        hex = rgb ?
+            (rgb[1] | 1 << 8).toString(16).slice(1) +
+            (rgb[2] | 1 << 8).toString(16).slice(1) +
+            (rgb[3] | 1 << 8).toString(16).slice(1) : orig;
+
+    if (alpha !== "") {
+        a = alpha;
+    } else {
+        a = '01';
     }
+    // multiply before convert to HEX
+    a = ((a * 255) | 1 << 8).toString(16).slice(1)
+    hex = hex + a;
 
-    return c;
+    return hex;
```

Comment:
Is there a reason this returns the value without the # symbol, could we rather do something like 

`return "#" + hex;`

Reply:
done 👍